CHANGELOG.md

```diff
@@ -4,6 +4,41 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
+## [3.16.0] - 2024-11-08
+
+### Changed
+
+* Drop support for PHP < 8.1, Symfony < 5.4 and Symfony 6.0 - 6.3. In future Behat will drop support for PHP and symfony
+  versions as they reach EOL. by @AlexSkrypnyk in [#1504](https://github.com/Behat/Behat/pull/1504)
+* ApplicationFactory::VERSION is deprecated and will not be updated, Behat now internally uses composer's runtime API
+  to report the running version. by @acoulton in [#1520](https://github.com/Behat/Behat/pull/1520)
+* API changes to 2 final Behat classes that are not considered part of the public API (but were not explicitly marked
+  as such). This may affect users who are creating instances direct rather than through the DI container as expected.
+  See Behat\Behat\EventDispatcher\Cli\StopOnFailureController in #1501 and Behat\Behat\Tester\Cli\RerunController in #1518.
+
+### Added
+
+* Render JUnit test durations with millisecond precision e.g. `1.234` rather than only as integer seconds
+  by @uuf6429 in [#1460](https://github.com/Behat/Behat/pull/1460)
+* Support specifying `stop_on_failure` within behat.yml by @jdeniau in [#1512](https://github.com/Behat/Behat/pull/1512),
+  [#1501](https://github.com/Behat/Behat/pull/1501) and [#1516](https://github.com/Behat/Behat/pull/1516)
+* Allow BeforeSuite/AfterSuite hooks to be marked with attributes by @rpkamp in [#1511](https://github.com/Behat/Behat/pull/1511)
+
+### Fixed
+
+* `--rerun` all tests that should be considered failed (including undefined, when strict) by @carlos-granados in [#1518](https://github.com/Behat/Behat/pull/1518)
+* Improve handling exceptions from unsupported PHPUnit versions by @acoulton in [#1521](https://github.com/Behat/Behat/pull/1521)
+* Fix high memory consumption when using Junit formatter by @simon-auch in [#1423](https://github.com/Behat/Behat/pull/1423)
+* Fix error when attempting to format long output messages by @jonpugh in [#1439](https://github.com/Behat/Behat/pull/1439)
+
+### Internal
+
+* Remove the unnecessary alias of the ScenarioInterface as it just causes confusion by @carlos-granados in [#1500](https://github.com/Behat/Behat/pull/1500)
+* Improve output when behat's own tests pass or fail unexpectedly by @jdeniau in [#1515](https://github.com/Behat/Behat/pull/1515)
+* Update guidance on submitting PRs by @acoulton in [#1505](https://github.com/Behat/Behat/pull/1505)
+* Fix indentation in GHA. by @AlexSkrypnyk in [#1502](https://github.com/Behat/Behat/pull/1502)
```

Comment:
I would probably spell GitHub Actions in full, otherwise some users may be left wondering what GHA is